### PR TITLE
feature/SP-167: Auto address resolution

### PR DESCRIPTION
**Pull request by AliAkiltayev**

### CODE REVIEW

File: client/src/main/java/io/mx51/spi/service/TenantService.java [hunk 1]
```diff
@@ -1,10 +1,12 @@
 package io.mx51.spi.service;
 
+import io.mx51.spi.model.DeviceAddressStatus;
```
Comment (RaymondK-mx51) [inline client/src/main/java/io/mx51/spi/service/TenantService.java:3]: I am not sure why we are referencing `DeviceAddressStatus` in `TenantService`?

File: client/build.gradle [hunk 2]
```diff
@@ -50,6 +50,7 @@ dependencies {
     implementation 'com.squareup.okhttp3:okhttp:3.12.3'
     implementation 'org.slf4j:slf4j-api:1.7.25'
     implementation 'org.jetbrains:annotations:16.0.1'
+    implementation 'commons-httpclient:commons-httpclient:3.1'
```
Comment (RaymondK-mx51) [inline client/build.gradle:53]: See my comment below

File: client/src/main/java/io/mx51/spi/service/AnalyticsService.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package io.mx51.spi.service;
 
 import com.google.gson.Gson;
+import io.mx51.spi.model.DeviceAddressStatus;
```
Comment (RaymondK-mx51) [inline client/src/main/java/io/mx51/spi/service/AnalyticsService.java:4]: I am not sure why we are referencing `DeviceAddressStatus` in `AnalyicsService`?

**Review by RaymondK-mx51 — CHANGES_REQUESTED** [review]
See my feedback, thanks

File: client/src/main/java/io/mx51/spi/util/DeviceHelper.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package io.mx51.spi.util;
+
+import com.google.gson.Gson;
+import io.mx51.spi.model.DeviceAddressResponseCode;
+import io.mx51.spi.model.DeviceAddressStatus;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.apache.commons.httpclient.HttpStatus;
```
[the remark this answers is not in the document]
Comment (AliAkiltayev) [reply client/src/main/java/io/mx51/spi/util/DeviceHelper.java:8]: okHttp has only http response code, but no description of this code.

**Review by RaymondK-mx51 — APPROVED** [review]

**Pull request merged**